Controls/Mail/Mailman.c

```diff
@@ -13,6 +16,26 @@
  */
 void depositLetter(void) {
 	OS_ERR err; // Make sure to check for errors and print the error code if not OS_ERR_NONE
+
+	// Generate Random Number
+	char randomNumberString[4];
+	sprintf(randomNumberString, "%d", rand() % 9999 + 1);
+
+	// Concatenate message to mailbox
+	strcpy(mailbox, "Get your own free pizza at ");
+	strcat(mailbox,  randomNumberString);
+	strcat(mailbox, " Gullible St!");
+
+	// Signal the recipient 
+	OSSemPost(&MailboxFlag_Sem4, OS_OPT_POST_1, &err);
+
+	// Print Finished!
+	printf("Finished!\n");
+
+	// Check for errors and print if present
+	if(err != OS_ERR_NONE) {
+		printf("depositLetter Error: %d\n", err);
+	}
```

Comment:
Probably wanna move to directly after the OSSemPost

Reply:
Resolved with new commit.